task_list_app/task_list_app/src/main/java/com/jackson/task_list_app/api/controller/TaskController.java

```diff
@@ -5,47 +5,86 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.jackson.task_list_app.api.models.Task;
+import com.jackson.task_list_app.api.models.TaskStatus;
 import com.jackson.task_list_app.api.services.TaskService;
 
-
-
 @RestController
+@CrossOrigin
 public class TaskController {
 
     @Autowired
     private TaskService taskService;
 
+    // Get Mappings
     @GetMapping("/tasks")
     @ResponseStatus(HttpStatus.OK)
-    public List<Task> getTask() {
-        return taskService.getTask();
+    public ResponseEntity<List<Task>> getTasks(@RequestParam(required = false) String status) {
+        if (status != null) {
+            try {
+                TaskStatus taskStatus = TaskStatus.valueOf(status.toUpperCase());
+                return new ResponseEntity<>(taskService.getTasksByStatus(taskStatus), HttpStatus.OK);
+            } catch (IllegalArgumentException e) {
+               return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+            }
+        } else {
+            return new ResponseEntity<>(taskService.getTask(), HttpStatus.OK);
+        }
     }
 
+    // Post Mappings
     @PostMapping("/tasks")
-    public ResponseEntity<Task> createTask(@RequestBody Task newTask) {
-        Task createdTask = taskService.createTask(newTask);
-        return new ResponseEntity<>(createdTask, HttpStatus.CREATED);
+    @ResponseStatus(HttpStatus.CREATED)
+    public Task createTask(@RequestBody Task newTask) {
+        return taskService.createTask(newTask);
     }
 
+    // Put Mappings
     @PutMapping("/tasks/{id}")
-    public ResponseEntity<Task> updateTask(@PathVariable Long id, @RequestBody Task updatedTask) {
+    public ResponseEntity<Task> updateTask(
+            @PathVariable Long id,
+            @RequestBody Task updatedTask,
+            @RequestParam(required = false) TaskStatus newStatus) {
```

Comment:
I don't think there should be a query parameter here. If you put a status in the request body, it looks like the query parameter is going to take precedence. I think the request should only give you one opportunity to update the status in the request body.

I think that should simplify the controller as well. All the logic that isn't error handling should probably exist in the service.

Reply:
I see this was addressed in the `card-401` branch